cli/tools/test_runner.rs

```diff
@@ -52,6 +53,141 @@ pub enum TestMessage {
   },
 }
 
+trait TestReporter {
+  fn visit_message(&mut self, message: TestMessage);
```

Comment:
Nitpick: would it make sense to take `&TestMessage` instead?

Reply:
Move is intentional, avoid having to deref every single field.

Okay with changing this, but then I'd prefer to make all reporters pass by ref.